### PR TITLE
Enable lazy lambda lifting with higher order logic

### PR DESCRIPTION
**Pull request by mudathirmahgoub**

### CODE REVIEW

File: src/theory/uf/ho_extension.cpp [hunk 1]
```diff
@@ -42,6 +42,12 @@ HoExtension::HoExtension(Env& env,
       d_uf_std_skolem(userContext())
 {
   d_true = NodeManager::currentNM()->mkConst(true);
+  if (d_env.getLogicInfo().isHigherOrder()
```
Comment (ajreynol) [inline src/theory/uf/ho_extension.cpp:45]: This is not the right place for this, options should never be modified after the SMT engine is initialized.

We should either move this code to `smt/set_defaults.cpp` or simply set the default value of the option to `true` in `options/uf_options`.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/theory/uf/ho_extension.cpp (hunk 1 above)
Comment (mudathirmahgoub) [reply src/theory/uf/ho_extension.cpp]: Modified options/uf_options

**Comment by ajreynol** [comment]
Can you comment on why the 2 regressions fail?

**Comment by ajreynol** [comment]
I will address this default option change in a separate PR.

**Pull request closed without merging**